model/registry/scm.yaml

```diff
@@ -0,0 +1,37 @@
+groups:
```

Comment:
In terms of ambiguity, confusion ...

How are you thinking about the difference between:

  - Attribution for the code that is currently being executed
  - Attribution for the code that is the subject of deployment

Suggestion that might make this clearer:

  - Put properties related to the currently executing code under "code.*"
     - https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/code.yaml

 - Put properties related to what you are deploying under a different prefix, perhaps something like `build.source.` or `deployment.source.`

For example:

 - `code.source.repo`: the repository of the code that is currently being executed (e.g. in the context of a CI/CD pipeline, this would be the code of the CI/CD pipeline itself as opposed to the code that the CI/CD pipeline is deploying)

 - `deployment.source.repo`: the repository of the code being deployed

Reply:
The object that contains a `build` will include an `scm` related only to that build, and the same is true for the deployment. I not sure it could happen, if both sections could be in the same object, if so, you do not know if the objects is about a build or a deployment.

Reply:
I generally agree with @kuisathaverat  but this is a good topic for another issue. I've opened #1210 to discuss this further outside of this pull request.